src/js/samples/components/__tests__/EditSample.test.tsx

```diff
@@ -0,0 +1,130 @@
+import { screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import React from "react";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+import { createFakeSample, mockApiEditSample } from "../../../../tests/fake/samples";
+import { renderWithProviders } from "../../../../tests/setupTests";
+import EditSample from "../EditSample";
+
+describe("<Editsample />", () => {
+    const sample = createFakeSample();
+    let props;
+
+    beforeEach(() => {
+        props = {
+            sample,
+            show: true,
+            onHide: vi.fn(),
+        };
+    });
+
+    it("should render when [show=false]", () => {
+        props.show = false;
+        renderWithProviders(<EditSample {...props} />);
+
+        expect(screen.queryByRole("textbox", { name: "Name" })).toBeNull();
+        expect(screen.queryByRole("textbox", { name: "Isolate" })).toBeNull();
+        expect(screen.queryByRole("textbox", { name: "Host" })).toBeNull();
+        expect(screen.queryByRole("textbox", { name: "Locale" })).toBeNull();
+        expect(screen.queryByRole("textbox", { name: "Notes" })).toBeNull();
+        expect(screen.queryByText("Save")).toBeNull();
+    });
+
+    it("should render after name is changed", async () => {
+        renderWithProviders(<EditSample {...props} />);
+
+        const nameInput = screen.getByLabelText("Name");
+        expect(nameInput).toBeInTheDocument();
+        expect(nameInput).toHaveValue(sample.name);
+
+        await userEvent.clear(nameInput);
+        expect(nameInput).toHaveValue("");
+
+        await userEvent.type(nameInput, "test");
+        expect(nameInput).toHaveValue("test");
+    });
+
+    it("should render after isolate is changed", async () => {
+        renderWithProviders(<EditSample {...props} />);
+
+        const isolateInput = screen.getByLabelText("Isolate");
+        expect(isolateInput).toBeInTheDocument();
+        expect(isolateInput).toHaveValue(sample.isolate);
+
+        await userEvent.clear(isolateInput);
+        expect(isolateInput).toHaveValue("");
+
+        await userEvent.type(isolateInput, "test");
+        expect(isolateInput).toHaveValue("test");
```

Comment:
I think there is room for improving these ` should render after x has changed` tests. Couple of ways I could see it going though.

Option 1:  Do the initial value check for all of them as part of a `should render` test then check the typing in the final `should submit` test. Would reduced test redundancy and having to render the dom as many times.

Option 2: you can probably use it.each since the tests are so similar.